docs/src/references/c8y-log-management.md

```diff
@@ -0,0 +1,122 @@
+On startup, it reports all the log file types that it manages, defined in the `c8y_log_plugin.toml`
+
+## Configuration
+
+The `c8y_log_plugin` configuration is stored by default under `/etc/tedge/c8y/c8y_log_plugin.toml`.
+
+This [TOML](https://toml.io/en/) file defines the list of log files that can be retrieved from the cloud tenant.
+The paths to these files can be represented using [glob](https://en.wikipedia.org/wiki/Glob_(programming)) patterns.
+The `type` given to these paths are used as the log type when they are reported to Cumulocity.
+
+```toml
+files = [
+    { type = "mosquitto", target = '/var/log/mosquitto.log' },
+    { type = "software-management", target = '/var/log/tedge/agent/software-management/software-*' },
+    { type = "c8y_LogRequest", target = '/var/log/tedge/agent/c8y_LogRequest/*' },
+    { type = "c8y_CustomOperation", target = '/var/log/tedge/agent/c8y_CustomOperation/*' }
+]
+```
+
+The `c8y_log_plugin` parses this configuration file on startup for all the `type` values specified,
+and sends the supported log types message(SmartREST `118`) to Cumuloicty on `c8y/s/us` topic as follows:
+
+```csv
+118,mosquitto,software-management,c8y_LogRequest,c8y_CustomOperation
+```
+
+The plugin continiously watches this configuration file for any changes and resends the `118` message with the `type`s in this file,
+whenever it is updated.
+
+Note: If the file `/etc/tedge/c8y/c8y_log_plugin.toml` is ill-formed or cannot be read,
+      then an empty `118` message is sent, indicating no log files are tracked.
+
+## Handling log requests from Cumulocity
+
+This plugin subscribes to `c8y/s/ds` topic, listening for `c8y_LogfileRequest` messages (SmartREST `522`) from Cumulocity, like this one:
+
+```csv
+522,<device-id>,mosquitto,2013-06-22T17:03:14.000+02:00,2013-06-22T18:03:14.000+02:00,ERROR,1000
+```
+
+The plugin then checks the `c8y_log_plugin.toml` file for the log type in the incoming message (`mosquitto`),
+retrives the log file using the `target` glob pattern provided in the plugin config file,
+filters the contents using the date range(`2013-06-22T17:03:14.000+02:00` to `2013-06-22T18:03:14.000+02:00`),
+the search text(`ERROR`) and the maximum line count(`1000`) and uploads this filtered content to Cumulocity.
+
+During this process, Cumulocity is notified of the progress of the `c8y_LogfileRequest` operation
+with SmartREST messages `501`(executing), `502`(failed) or `503`(successful) on `c8y/s/us` topic.
+
+## Updating supported log files from Cumulocity
+
+The supported log files list defined in the `c8y_log_plugin.toml` can be updated both locally as well as from Cumulocity cloud.
+Updates from Cumulocity can be achieved simply by listing this log plugin's confg file(`c8y_log_plugin.toml`) 
+in the configuration file of the `c8y_configuration_plugin`.
+This will enable the `c8y_log_plugin.toml` to be tracked and managed by the `c8y_configuration_plugin`.
+
+## Usage
+
+```shell
+$ c8y_log_plugin
```

Comment:
It might be good to add an `--init` function that creates the `/etc/tedge/operations/c8y/c8y_LogfileRequest` file as this as been done for the configuration management plugin.